code/modules/mob/living/simple_animal/hostile/thing.dm

```diff
@@ -0,0 +1,2046 @@
+	var/mob/living/simple_animal/hostile/thing/user = owner
+	user.armbladetransition()
+
+/datum/action/innate/thing/communicate
+	name = "Communicate"
+	button_icon_state = "thing_communicate"
+	desc = "Communicate with your spawn."
+
+/datum/action/innate/thing/communicate/Activate()
+	var/mob/living/simple_animal/hostile/thing/user = owner
+	user.UndefinedCommunication()
+
+/datum/action/innate/thing/screech
+	name = "Screech"
+	button_icon_state = "thing_screech"
+	desc = "Compromise any living being's motor skills with an impossibly disorientating noise."
+
+/datum/action/innate/thing/screech/Activate()
+	var/mob/living/simple_animal/hostile/thing/user = owner
+	user.disorientatingscreech()
+
+/datum/action/innate/thing/empscreech
+	name = "Electromagnetic Disturbance"
+	button_icon_state = "thing_shockscreech"
+	desc = "Compromise machines with an electric blast."
+
+/datum/action/innate/thing/empscreech/Activate()
+	var/mob/living/simple_animal/hostile/thing/user = owner
+	user.EMP()
+
+/datum/action/innate/thing/plantcreep
+	name = "Build - Lay Creep"
+	button_icon_state = "thing_plantcreep"
+	desc = "Cover the surrounding area in creep."
+
+/datum/action/innate/thing/plantcreep/Activate()
+	var/mob/living/simple_animal/hostile/thing/user = owner
+	user.Spawncreepnode()
+
+/datum/action/innate/thing/build
+	name = "Build - Create Building"
+	button_icon_state = "thing_build"
+	desc = "Create a useful building on creep."
+
+/datum/action/innate/thing/build/Activate()
+	var/mob/living/simple_animal/hostile/thing/user = owner
+	user.Spawncreepbuilding()
+
+/datum/action/innate/thing/crawlerrally
+	name = "Rally Crawlers"
+	button_icon_state = "thing_crawlerrally"
+	desc = "Rally your crawlers towards you."
+
+/datum/action/innate/thing/crawlerrally/Activate()
+	var/mob/living/simple_animal/hostile/thing/user = owner
+	user.rally_crawlers()
+
+/datum/action/innate/thing/evolve
+	name = "Evolve"
+	button_icon_state = "thing_evolve"
+	desc = "Evolve into a stronger being."
+
+/datum/action/innate/thing/evolve/Activate()
+	var/mob/living/simple_animal/hostile/thing/user = owner
+	user.Evolve()
+
+// Structures
+
+// Creep, practically just reskinned weeds
+
+/obj/structure/alien/creep
+	gender = PLURAL
+	name = "creep"
+	desc = "Intertwined tentacles are covering the floor... this can't be good."
+	anchored = TRUE
+	density = FALSE
+	layer = TURF_LAYER
+	plane = FLOOR_PLANE
+	icon = 'icons/mob/mob.dmi'
+	icon_state = "creep"
+	max_integrity = 30
+	var/obj/structure/alien/creep/node/linked_node = null
+	var/static/list/creepImageCache
+
+
+/obj/structure/alien/creep/Initialize(pos, node)
+	..()
+	linked_node = node
+	if(istype(loc, /turf/space))
+		qdel(src)
+		return
+	fullUpdateCreepOverlays()
+	spawn(rand(150, 200))
+		if(src)
+			Life()
+
+/obj/structure/alien/creep/Destroy()
+	for(var/obj/structure/alien/creep/W in range(1,loc))
+		W.updateWeedOverlays()
+	linked_node = null
+	return ..()
+
+/obj/structure/alien/creep/proc/Life()
```

Comment:
Yo what. Structure doesn't have life() defining it here won't make it run. What the.

it is process()